### DISCUSSION TITLE
v1.2.0 Proposals - General Fixes and Error Handling

### DISCUSSION THREAD

**Discussion by mochi-co** [mochi-mqtt/server#48] [Ideas]
**v1.2.0** will be ready for release shortly and as `mochi-co/mqtt` is transitioning from a solo project to a multi-contributor project (🥳 ), I thought it would be a good idea to get a bit more organised.

### v1.2.0 contains:
- https://github.com/mochi-co/mqtt/pull/29 (@jmacd)
- https://github.com/mochi-co/mqtt/pull/41 (@mochi-co)
- https://github.com/mochi-co/mqtt/pull/57 (@deadprogram)
- https://github.com/mochi-co/mqtt/pull/53 (@stffabi)
- https://github.com/mochi-co/mqtt/pull/63 (@mochi-co)
- https://github.com/mochi-co/mqtt/pull/61 (@mochi-co)
- **Improve error handling:**
    - https://github.com/mochi-co/mqtt/pull/38 (@jmacd)
    - https://github.com/mochi-co/mqtt/pull/39 (@jmacd)
- **Bug fixes:**
    - https://github.com/mochi-co/mqtt/pull/67 (@mochi-co)
    - https://github.com/mochi-co/mqtt/pull/36 @jmacd)
    - https://github.com/mochi-co/mqtt/pull/42 (@stffabi)
    - https://github.com/mochi-co/mqtt/pull/46 (@stffabi)
    - https://github.com/mochi-co/mqtt/pull/51 (@jmacd)
    - https://github.com/mochi-co/mqtt/pull/58 (@soyoo)
    - https://github.com/mochi-co/mqtt/pull/62 (@mochi-co)
    - https://github.com/mochi-co/mqtt/pull/68 (@mochi-co)

**Comment by mochi-co** [comment]
I think it would be nice to resolve the remaining issues
- https://github.com/mochi-co/mqtt/issues/21
- https://github.com/mochi-co/mqtt/issues/43 
- https://github.com/mochi-co/mqtt/issues/44 

as part of v1.2.0, as this would wrap it up nicely.

**Comment by jmacd** [comment]
Does the change in #38 constitute a breaking API change related to the change of `events.Client` to `events.ClientLike`?
I might just-to-be-safe call it 1.2.0, then use 1.3 for the redis changes.

**Reply by mochi-co** [reply]
I agree - I have renamed it to 1.2.0. Redis persistence can be added as a future minor version. I think clustering needs more serious architectural thought before implementation.
